custom_components/tahoma/climate_devices/atlantic_pass_apcdhw.py

```diff
@@ -0,0 +1,178 @@
+COMMAND_REFRESH_ECO_TARGET_DWH_TEMPERATURE = "refreshEcoTargetDHWTemperature"
+COMMAND_REFRESH_TARGET_DWH_TEMPERATURE = "refreshTargetDHWTemperature"
+COMMAND_SET_BOOST_ON_OFF_STATE = "setBoostOnOffState"
+COMMAND_SET_COMFORT_TARGET_DWH_TEMPERATURE = "setComfortTargetDHWTemperature"
+COMMAND_SET_DWH_ON_OFF_STATE = "setDHWOnOffState"
+COMMAND_SET_ECO_TARGET_DWH_TEMPERATURE = "setEcoTargetDHWTemperature"
+COMMAND_SET_PASS_APCDHW_MODE = "setPassAPCDHWMode"
+
+
+CORE_BOOST_ON_OFF_STATE = "core:BoostOnOffState"
+CORE_COMFORT_TARGET_DWH_TEMPERATURE_STATE = "core:ComfortTargetDHWTemperatureState"
+CORE_DHW_DEROGATION_AVAILABILITY_STATE = "core:DHWDerogationAvailabilityState"
+CORE_DWH_ON_OFF_STATE = "core:DHWOnOffState"
+CORE_ECO_TARGET_DWH_TEMPERATURE_STATE = "core:EcoTargetDHWTemperatureState"
+CORE_STATUS_STATE = "core:StatusState"
+CORE_TARGET_DWH_TEMPERATURE_STATE = "core:TargetDHWTemperatureState"
+
+CUSTOM_PRESET_OFF = "Off"
+CUSTOM_PRESET_PROG = "Prog"
+
+DWH_OFF_STATE = "off"
+DHW_ON_STATE = "on"
+
+IO_PASS_APCDHW_CONFIGURATION_STATE = "io:PassAPCDHWConfigurationState"
+IO_PASS_APCDWH_MODE_STATE = "io:PassAPCDHWModeState"
+IO_PASS_APCDHW_PROFILE_STATE = "io:PassAPCDHWProfileState"
+
+PASS_APCDHW_MODE_COMFORT = "comfort"
+PASS_APCDHW_MODE_ECO = "eco"
+PASS_APCDWH_MODE_INTERNAL_SCHEDULING = "internalScheduling"
+PASS_APCDHW_MODE_STOP = "stop"
+
+
+MAP_HVAC_MODES = {
+    DHW_ON_STATE: HVAC_MODE_HEAT,
+    DWH_OFF_STATE: HVAC_MODE_OFF,
+}
+MAP_REVERSE_HVAC_MODES = dict(map(reversed, MAP_HVAC_MODES.items()))
+
+MAP_PRESET_MODES = {
+    PASS_APCDHW_MODE_ECO: PRESET_ECO,
+    PASS_APCDHW_MODE_COMFORT: PRESET_COMFORT,
+    PRESET_BOOST: PRESET_BOOST,
+    PASS_APCDWH_MODE_INTERNAL_SCHEDULING: CUSTOM_PRESET_PROG,
+    PASS_APCDHW_MODE_STOP: CUSTOM_PRESET_OFF,
+}
+MAP_REVERSE_PRESET_MODES = dict(map(reversed, MAP_PRESET_MODES.items()))
+
+
+class AtlanticPassAPCDHW(TahomaDevice, ClimateEntity):
+    """Representation of TaHoma IO Atlantic Electrical Heater."""
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def min_temp(self) -> float:
+        """Return minimum temperature."""
+        return 30
+
+    @property
+    def max_temp(self) -> float:
+        """Return maximum temperature."""
+        return 65
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self.select_state(IO_PASS_APCDWH_MODE_STATE) in [
+            PASS_APCDHW_MODE_ECO,
+            PASS_APCDWH_MODE_INTERNAL_SCHEDULING,
+            PASS_APCDHW_MODE_STOP,
+        ]:
+            return MAP_PRESET_MODES[self.select_state(IO_PASS_APCDWH_MODE_STATE)]
+        else:
```

Comment:
Useless else. You have a return in the previous if.

Reply:
fixed